packages/apollos-data-connector-onesignal/src/data-source.js

```
@@ -24,6 +24,7 @@ export default class OneSignal extends RESTDataSource {
     subtitle,
     ...args
   }) {
+    if (!ONE_SIGNAL?.APP_ID) return null;
```

Should this be one of those areas where we fail harder rather than hiding our errors? As in - if we aren;t using `OneSignal`, shouldn't we opt out of the entire datasource? We've launched an app or two in the past with no `ONE_SIGNAL.APP_ID` accidentally and it's caused a headache tracking it down.

---

yes I agree, but it wasn't failing gracefully in this case, posting a journal was just spinning and never resolving. I think my "no config yaml" PR can help tremendously with this problems we've seen because it checks for the appropriate env variables on module import

---

I guess I'm advocating that we don't want a graceful failing here :) We should bomb hard so we avoid launching without OneSignalID, or remove this datasource entirely.

---

I agree, what do you think about my "no config yaml" PR given this? I think it eliminates a lot of confusion and makes it the server harder to misuse

---

This is a discussion for that PR, but I don't like how much configuration is hidden when we take out the `config.yml`. Even just the three code paths we have right now that are managed by config.yml-less configuration have been confusing to manage while developing, and I'm really worried about making it worse